common/src/main/java/net/adoptopenjdk/icedteaweb/jvm/JVMUtils.java

```diff
@@ -0,0 +1,113 @@
+        boolean argumentIsValid;
+        for (String argument : arguments) {
+            argumentIsValid = false;
+
+            if (validArguments.contains(argument)) {
+                argumentIsValid = true;
+            } else {
+                for (String validStartingArgument : validStartingArguments) {
+                    if (argument.startsWith(validStartingArgument)) {
+                        argumentIsValid = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!argumentIsValid) {
+                throw new IllegalArgumentException(argument);
+            }
+        }
+    }
+
+    /**
+     * Returns an array of valid (ie safe and supported) arguments for the JVM
+     *
+     * Based on
+     * http://java.sun.com/javase/6/docs/technotes/guides/javaws/developersguide/syntax.html
+     */
+    private static String[] getValidVMArguments() {
+        return new String[]{
+            "-d32", /* use a 32-bit data model if available */
+            "-client", /* to select the client VM */
+            "-server", /* to select the server VM */
+            "-verbose", /* enable verbose output */
+            "-version", /* print product version and exit */
+            "-showversion", /* print product version and continue */
+            "-help", /* print this help message */
+            "-X", /* print help on non-standard options */
+            "-ea", /* enable assertions */
+            "-enableassertions", /* enable assertions */
+            "-da", /* disable assertions */
+            "-disableassertions", /* disable assertions */
+            "-esa", /* enable system assertions */
+            "-enablesystemassertions", /* enable system assertions */
+            "-dsa", /* disable system assertione */
+            "-disablesystemassertions", /* disable system assertione */
+            "-Xmixed", /* mixed mode execution (default) */
+            "-Xint", /* interpreted mode execution only */
+            "-Xnoclassgc", /* disable class garbage collection */
+            "-Xincgc", /* enable incremental garbage collection */
+            "-Xbatch", /* disable background compilation */
+            "-Xprof", /* output cpu profiling data */
+            "-Xdebug", /* enable remote debugging */
+            "-Xfuture", /* enable strictest checks, anticipating future default */
+            "-Xrs", /* reduce use of OS signals by Java/VM (see documentation) */
+            "-XX:+ForceTimeHighResolution", /* use high resolution timer */
+            "-XX:-ForceTimeHighResolution", /* use low resolution (default) */};
+    }
+
+    /**
+     * Returns an array containing the starts of valid (ie safe and supported)
+     * arguments for the JVM
+     *
+     * Based on
+     * http://java.sun.com/javase/6/docs/technotes/guides/javaws/developersguide/syntax.html
+     */
+    private static String[] getValidStartingVMArguments() {
```

Comment:
could be converted in a constant

Reply:
according to zoom session: we do that in a separate PR